prov/tcp/src/tcpx_init.c

```diff
@@ -58,6 +61,13 @@ static void tcpx_getinfo_ifs(struct fi_info **info)
 	for (ifa = ifaddrs; ifa != NULL; ifa = ifa->ifa_next) {
 		if (ifa->ifa_addr == NULL || !(ifa->ifa_flags & IFF_UP))
 			continue;
+		if (tcpx_interface_name &&
+		    (strncmp(tcpx_interface_name, ifa->ifa_name,
+			     strlen(tcpx_interface_name)) != 0)) {
+			FI_DBG(&tcpx_prov, FI_LOG_CORE,
+			       "Skip (%s) interface\n", ifa->ifa_name);
+			continue;
```

Comment:
If there is no match for the tcpx_interface_name, I assume that we need to return NULL in *info. Is that the expectation?

if yes, we have a problem. Down in this function, we return the *info as-is, when we don't find any matches. Wouldn't it be NULL?

Reply:
My expectation would be if no matching interfaces are found, we return zero fi_info structures.

Reply:
I updated the PR. Please, check that it addresses everything that you requested.
Now it returns `*info = NULL` to the userm if the `FI_TCP_IFACE` is set to unknown iface name.